### ISSUE TITLE
@ApiModelProperty not displaying attribute when using @ModelAttributeode

### ISSUE THREAD

**Issue by rantunesboreas** [springfox/springfox#1449]
Hi, 

I have a POJO (FilterRequest) that is used to configure various filter parameters for submission to a REST endpoint. I have added @ModelAttribute in the controller endpoint so that it is picked up by springfox.

``` java
@ApiOperation(value = "Filtering endpoint", produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<<FilterResponse>> search(@ModelAttribute @Valid FilterRequest filterRequest) {
   ...
}

//The FilterRequest is defined as follows:

public class FilterRequest {
   private static final String FILTER1_NAME = "filter1"
   ...
   private static final String FILTER_N_NAME = "filterN"

   private final Map<String, String> filterMap = new HashMap<>();

   public void setFilter1(String filter){
       if (filter != null) {
            filterMap.put(FILTER1_NAME, filter);
        }
   }
   @ApiModelProperty(value = "Filters by filter1", required = true)
   public String getFilter1(){filterMap.get(FILTER1_NAME);}

    ...
   public void setFilterN(String filter){...}

   @ApiModelProperty(value = "Filters by filterN", required = true)
   public String getFilterN(){...}
}
```

In principle this should be enough seeing as @ApiModelProperty should work on getters. However this is not the case, and nothing appears in the swagger output. If I declare the filters as individual state attributes they appear. I want to be able to save all these fields in a map, seeing as I need to process these down the line.

How can I get the @ApiModelProperty to wotk in my case>?

**Comment by dilipkrish** [comment]
I think it should be on the setters and not the getters.

**Comment by rantunesboreas** [comment]
HI Dilip,

Thanks for replying. I have tried applying the @ApiModelProperty to the setter methods but with no success. The filters still don't appear in the documentation.

Do you have any other suggestions?

**Comment by dilipkrish** [comment]
Let me take a look

**Comment by dilipkrish** [comment]
Yeah this is a limitation that only properties backed by fields are rendered. Sorry I glossed over that when I first saw this issue. The reason for this is that we wanted a way to deterministically display only the attributes that were physically stored. 

I've changed this story to a feature, Would be great if you would help out with a PR!

**Comment by rantunesboreas** [comment]
Sure, I'll give it a try once I get back from holiday.

**Comment by dilipkrish** [comment]
Fixed via #1937
